### ISSUE TITLE
Validation of attribute constraints occurs prior to custom methods execution, allows non-unique data

### ISSUE THREAD

**Issue by eric-schleicher** [structr/structr#360]
I just noticed that if I set a column to be unique, and create a method that modifies the value in that column, I can violate the attribute constraints.

I believe this may be because validation occurs prior to method execution, which seems not so good from a data management perspective (allow violation of Consistency principle  A**C**ID)

Here is a representative example:

-Object: TestObject
 -Attributes: 
 - name (not null, unique)
 - properName (not null)

methods:
onCreate: 
```
(
set(this, 'name', clean(this.nameProper))
)
```

onSave:
```
(
set(this, 'name', clean(this.nameProper))
)
```

When adding a new record in #Data, the modal dialog pop up to take an entry (since there are not nullable attributes)

if I leave the name column empty, the form validation prevents completion (understandable, since the form has no idea that the methods will be modifying the value, but IMHO should try).

if I enter any value (just to get the form to accept) i notice the following, the submission occurs, but produces the following 

##### Observation.

The function ```clean()``` will result in character mapping a bunch of characters to the '-' (hyphen) to make it url friendly., this means that the the following string example result in the same output:
1. clean('this is a string') = 'this-is-a-string'
2. clean('this_is-a string') = 'this-is-a-string'
3. clean('this-is a-string') = 'this-is-a-string'

All three of the inputs were allowed

##### Expected result:
With the name attribute defined as a not null / unique attribute, records entered with these values should be expected to fail (2., 3.), since the resulting value (after the methods run) of the name attribute is in fact not unique.  otherwise, one would have to write their own validation logic to pre-check and manually raise the error to prevent non-deterministic results name on value in the name attribute.

**Comment by cmorgner** [comment]
This issue should be fixed with commit 13804fda8dbe7abb16f88f58320ae419b8bd1027. There is not yet a new snapshot that contains the fix.

**Comment by amorgner** [comment]
@eric-schleicher This issue should be fixed since a couple of months. If you're still seeing issues here, feel free to re-open this issue, thanks!
